nixos/modules/config/users-groups.nix

```diff
@@ -588,7 +593,7 @@ in {
              || cfg.group == "wheel"
              || elem "wheel" cfg.extraGroups)
             &&
-            ((cfg.hashedPassword != null && cfg.hashedPassword != "!")
+            (cfg.hashedPassword != null
```

Comment:
I’m super uncomfortable with this line, somebody else will have to confirm that it’s valid.

Reply:
There are two possible cases I can think of:
- using the deprecated `security.initialRootPassword` "!" will be turned into a null, so it's covered
- using `users.users.<name>.hashedPassword` or similar, "!" will issue the warning about possibly being locked out.

But yes, it would be better is someone familiar with this module would comment.

Reply:
Looks to me like @edolstra is the only one who touched this code in the past.

Reply:
Yeah, user using `users.users.<name>.hashedPassword` that is not a hash will lock themselves out if they miss the warning so the assertion still makes sense. It is even too narrow here – I use `*` as a root’s hash and this would not save me. Not sure we can extend it much further without causing false positives, though.